storage/storagepb/kv.proto

```diff
@@ -14,11 +14,14 @@ message KeyValue {
   int64 create_revision = 2;
   // mod_revision is the last modified revision of the key.
   int64 mod_revision = 3;
+  // tombstone_revision is the tombstone revision of the key.
+  // If the key is not tombstoned, its revision is set to zero.
+  int64 tombstone_revision = 4;
```

Comment:
> Why this cannot be done in watch layer?

It belongs to key-value struct IMO.

> I think this is exact mod_revision. And we should just use mod_revision.

I think these two ways are similar. If you really prefer to use mod_revision and change tombstone to bool var, I could make the change.